tsinfer/formats.py

```diff
@@ -74,6 +74,43 @@ def permissive_json_schema():
     }
 
 
+def node_metadata_schema():
+    # This is fixed by tsinfer: users cannot add to the node metadata
+    return tskit.MetadataSchema(
+        {
+            "codec": "struct",
+            "type": ["object", "null"],
+            "properties": {
+                "tsinfer": {
+                    "description": "Information about node identity "
+                    "from the tsinfer inference process",
+                    "type": "object",
+                    "default": {"ancestor_data_id": -1, "sample_data_id": -1},
```

Comment:
What's this default for? I'm not sure we can do this.

Reply:
This is needed because if a top level object is not marked as "required", it needs a top level default. This does work, and I've checked it with @benjeffery, although perhaps he could verify?

Reply:
Thinking about it, I'm not sure this is what you want. For the lat/lng case it made sense - doing the default in this way says "Either both properties are the default or both are specified". For this case I think you want them to be independent, and the default should be specified separately at the deeper level.

Reply:
I just thought this through and indeed we can delete the higher-level default, I think. I've updated the PR.

Reply:
Yes, the proprty becomes required - but it can just be an empty object.